packages/moonstone/Scrollable/ScrollButtons.js

```diff
@@ -286,46 +251,62 @@ class ScrollButtons extends Component {
 				if (focusableScrollButtons && !Spotlight.getPointerMode()) {
 					preventDefault(ev);
 					Spotlight.setPointerMode(false);
-					Spotlight.focus(ReactDOM.findDOMNode(this.nextButtonRef.current)); // eslint-disable-line react/no-find-dom-node
-				} else if (!nextButtonDisabled) {
+					Spotlight.focus(ReactDOM.findDOMNode(oppositeButtonRef)); // eslint-disable-line react/no-find-dom-node
+				} else if (!oppositeButtonDisabled) {
 					preventDefault(ev);
-					this.onClickNext(ev);
+					oppositeButtonClick(ev);
 				}
-			} else if (!prevButtonDisabled) {
+			} else if (!buttonDisabled) {
 				preventDefault(ev);
 				this.onClickPrev(ev);
 			}
-		}
-	}
-
-	onKeyDownNext = (ev) => {
-		const
-			{focusableScrollButtons, vertical} = this.props,
-			{nextButtonDisabled, prevButtonDisabled} = this.state,
-			{keyCode} = ev;
-
-		if (isPageDown(keyCode) || isPageUp(keyCode)) {
-			if (!vertical) {
-				ev.preventDefault();
-				return;
-			}
+		} else if (direction) {
+			const
+				{rtl} = this.props,
+				isDown = direction === 'down',
+				isLeftMovement = direction === (rtl ? 'right' : 'left'),
+				isRightMovement = direction === (rtl ? 'left' : 'right'),
+				isUp = direction === 'up',
+				isNextButton = position === 'next',
+				isPrevButton = position === 'prev',
+				fromNextToPrev = vertical ? isUp : isLeftMovement,
+				fromPrevToNext = vertical ? isDown : isRightMovement;
 
-			if (isPageUp(keyCode)) {
-				if (focusableScrollButtons && !Spotlight.getPointerMode()) {
-					preventDefault(ev);
-					Spotlight.setPointerMode(false);
-					Spotlight.focus(ReactDOM.findDOMNode(this.prevButtonRef.current)); // eslint-disable-line react/no-find-dom-node
-				} else if (!prevButtonDisabled) {
+			// manually focus the opposite scroll button when 5way pressed
+			if (isNextButton && fromNextToPrev || isPrevButton && fromPrevToNext) {
+				this.focusOnOppositeScrollButton(ev, direction);
+			} else {
+				const
+					// If it is vertical `Scrollable`, move focus to the left for ltr or to the right for rtl
+					// If is is horizontal `Scrollable`, move focus to the up
+					directionToContent = !vertical && 'up' || rtl && 'right' || 'left',
+					isDirectionToLeave =
+						vertical && (isRightMovement || isPrevButton && isUp || isNextButton && isDown) ||
+						!vertical && (isDown || isPrevButton && isLeftMovement || isNextButton && isRightMovement);
+
+				if (isDirectionToLeave) {
+					if (!getTargetByDirectionFromElement(direction, ev.target)) {
+						// move focus into contents and stop bubbling
+						preventDefault(ev);
```

Comment:
> In this case, we wouldn't want to prevent event bubbling. Since we're moving in a "leaving" direction and there is no target in the direction pressed, that means focus is attempting to leave the scroller container area. In that case, an application should be able to receive onKeyDown event passed to VirtualList.

As far as I know, if there is no target to move when press 5-way key on non-focusable scrollbar buttons, focus should move into contents and it means the key input is consumed. (Well, I missed to check focusableScrollbar here. It is a bug. :() So, preventDefault() is intended here. Is this rule different between `Scroller` and `VirtualList`?

> One problem you will have with this implementation is that there should be a difference in whether we prevent onKeyDown events based on whether a Scroller is being used vs a VirtualList.

Thank you, you got the point about what is missing between us! I didn't know any requirements to bubble events to apps from `Scroller`, not like `VirtualList`.